### PR TITLE
feat(solana): Send Solana Tokens

### PR DESCRIPTION
**Pull request by memoriesadrift**
~~**The PR is a bit messy because it's based on two branches. It will clear up a lot once #9496 and #9553 are merged.**~~
~~Until then, review starting from `6edcd0998a7c4b2c11b345d32df5d5c0deefe352`~~
Tested on firmware commit: `a8c2b7df7e79b0ece5915f5d6028963597085099`

## Description
This PR introduces sending arbitrary SPL tokens (blanket Solana term for FTs and NFTs like ERC 20, ERC 721, ERC 1155...)  
Users can send tokens to arbitrary accounts, creating associated token accounts for receivers if need be.

### How tokens work on Solana
Tokens on Solana are handled by creating auxillary accounts known as (associated) token accounts, each responsible for holding a single SPL token. A user can have an arbitrary amount of token accounts, including an arbitrary amount for a single token. Token accounts can be derived from the user's base address, known as *associated token accounts*, and this is the preferred way of holding tokens. The most popular Solana wallets create and manage these associated token accounts under the hood, not exposing them to users. *We've chosen this approach as well.*
To account for powerusers and special use cases, we collect the user's token balances across all their accounts holding the given token and display this aggregated balance. When sending tokens, we collect tokens as necessary from across these accounts.  
A transaction sending tokens transfers funds from the user's token account(s) to the recipient's associated token account, derived from the recipient's base address that the sender provides. This can be overriden by providing a token account address directly in the send modal. If the recipient doesn't have an associated token account for the given token, we attach an instruction to create that account as well as fund its [rent-exempt status](https://docs.solana.com/developing/programming-model/accounts#rent-exemption) (this is the [de facto best practice](https://spl.solana.com/associated-token-account#motivation)).  
Sending tokens to a token account designated for holding a different token isn't checked by suite -- instead the transaction fails gracefully when submitted with the error message `Transaction signing error: failed to send transaction: Transaction simulation failed: Error processing Instruction 0: custom program error: 0x3`. No funds are consumed and this only happens if the user tries to manually send funds to a token account address holding a different token, which is considered a rare use case.  
When receiving tokens, the user can provide their base address to the sender, though how the sending of tokens is handled is up to the sender's wallet. We don't believe it's worthwhile to display the individual token accounts' addresses to the user. If need be, they are visible in the explorer ([example](https://explorer.solana.com/address/ETxHeBBcuw9Yu4dGuP3oXrD12V5RECvmi8ogQ9PkjyVF/tokens?cluster=devnet&display=detail)).

### CODE REVIEW

File: packages/connect/src/data/defaultFeeLevels.ts [hunk 1]
```diff
@@ -109,11 +109,20 @@ const CARDANO_FEE_INFO: FeeInfoWithLevels = {
     dustLimit: -1, // unknown/unused
 };
 
+const SOLANA_FEE_INFO: FeeInfoWithLevels = {
+    blockTime: -1, // unknown
+    defaultFees: [{ label: 'normal', feePerUnit: '5000', blocks: -1 }],
+    minFee: 5000,
+    maxFee: 100000000, // arbitrary
```
Comment (memoriesadrift) [inline packages/connect/src/data/defaultFeeLevels.ts:116]: Solana fees can essentially be arbitrary. How should we handle this?
Right now, the highest fees we can produce are `2044280` from transactions creating associated token accounts, [example](https://explorer.solana.com/address/ETxHeBBcuw9Yu4dGuP3oXrD12V5RECvmi8ogQ9PkjyVF?cluster=devnet). The price for creating token accounts is not fixed and could technically change, same as the base fee. It hasn't in the history of the chain, but it *may*.

**Comment by mroz22** [comment]
conflicts here, please rebase 🙏

File: packages/blockchain-link/package.json [hunk 2]
```diff
@@ -67,6 +67,7 @@
         "worker-loader": "^3.0.8"
     },
     "dependencies": {
+        "@solana/buffer-layout": "^4.0.1",
```
Comment (mroz22) [inline packages/blockchain-link/package.json:70]: hmm good for feat/solana but might be questioned later. can't we handle this with current deps?

File: packages/connect/src/data/defaultFeeLevels.ts (hunk 1 above)
Comment (mroz22) [reply packages/connect/src/data/defaultFeeLevels.ts:116]: 🤷‍♂️

**Review by mroz22 — APPROVED** [review]

**Pull request merged**